### PR TITLE
Client: Bump shared eslint config and remove no-null/no-null plugin

### PR DESCRIPTION
**Pull request by tylerbutler**
Includes the config changes from #8978 and #8980.

### CODE REVIEW

File: packages/test/test-end-to-end-tests/src/test/sharedInterval.spec.ts [hunk 1]
```diff
@@ -654,8 +650,7 @@ describeFullCompat("SharedInterval", (getTestObjectProvider) => {
 
                 intervals1.changeProperties(id1, { prop1: "maybe" });
                 assertPropertyChangedArg(deltaArgs1.prop1, "yes", "Mismatch in property-changed event arg 9");
-                await provider.opProcessingController.processOutgoing();
-                // eslint-disable-next-line no-null/no-null
+        await provider.opProcessingController.processOutgoing();
```
Comment (heliocliu) [inline packages/test/test-end-to-end-tests/src/test/sharedInterval.spec.ts:653]: nit: looks like you lost a tab here

**Review by heliocliu — APPROVED** [review]
just fyi it looks like client hasn't taken the current server version yet but will be forced to do so as part of the next release.  should be fine as long as server isn't holding latent breaking changes 🤷‍♂️

**Comment by heliocliu** [comment]
what happened with the server changes here?  did you revert them because of the cpp compile errors in the logs?  it looked like there were some normal no-null errors at the end of that log - not sure if that was what was failing the build though (despite the other stuff)

**Comment by tylerbutler** [comment]
@heliocliu Yeah I realized I'd missed some stuff in server, then decided to split it out into another change to try and expedite the client changes. These big changes are so hard to keep current. But now I am working through CI test failures -- mostly due to oversights on my part.

**Pull request merged**